tests/environment.py

```diff
@@ -336,3 +349,83 @@ def launch_environment_docker(unique_id: str, skip_init: bool = False, tag: str
     debug('agent2: {}'.format(agents[1].mac))
     debug('agent2 wlan0: {}'.format(agents[1].radios[0].mac))
     debug('agent2 wlan2: {}'.format(agents[1].radios[1].mac))
+
+
+class ALEntityPrplWrt(ALEntity):
+    """Abstraction of ALEntity in real device."""
+
+    def __init__(self, name: str, device: None, is_controller: bool = False):
+        self.device = device
+        self.name = name
+        self.bridge_name = 'br-lan'
+
+        if is_controller:
+            self.config_file_name = '/opt/prplmesh/config/beerocks_controller.conf'
+        else:
+            self.config_file_name = '/opt/prplmesh/config/beerocks_agent.conf'
+
+        ucc_port = self.command(("grep \"ucc_listener_port\" {} "
+                                "| cut -d'=' -f2 | cut -d\" \" -f 1").format(self.config_file_name))
+
+        device_ip_output = self.command(
+                "ip -f inet addr show {} | head -n 2".format(self.bridge_name))
```

Comment:
`head` is just to get last of IPs, since `ip` may report multiple IPs for same interface. E.g :
```
root@OpenWrt:/# ip -family inet addr show br-lan
13: br-lan: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP qlen 1000
    inet 192.168.1.1/24 brd 192.168.1.255 scope global br-lan
       valid_lft forever preferred_lft forever
    inet 192.168.250.170/24 brd 192.168.250.255 scope global br-lan
       valid_lft forever preferred_lft forever
    inet 172.21.0.245/16 scope global br-lan
       valid_lft forever preferred_lft forever
```

But with command like this, we may get IPv6 address instead of IPv4, which is not expected. So it better to change it to:
`ip -family inet addr show {} | head -n 2`

Reply:
I'll repeat my statement: since you do a regex search below, head is redundant. re.search will return the first match, so whether you do the `head` or not, you get the same result.